processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -214,7 +214,26 @@ public File mergeQueryableIndex(
     return mergeQueryableIndex(indexes, rollup, metricAggs, outDir, indexSpec, new BaseProgressIndicator());
   }
 
-  public File mergeQueryableIndex(
+  public File mergeAndCloseMergee(
+      List<QueryableIndex> indexes,
+      boolean rollup,
+      final AggregatorFactory[] metricAggs,
+      File outDir,
+      IndexSpec indexSpec,
+      ProgressIndicator progress
+  ) throws IOException
+  {
+    try {
+      return mergeQueryableIndex(indexes, rollup, metricAggs, outDir, indexSpec, progress);
+    }
+    finally {
```

Comment:
Hmm, we used to have two versions of `mergeQueryableIndex` that behaved the same, except one let callers specify a progress indicator. With this patch, now the no-progress-indicator version keeps inputs open, but the progress-indicator version closes inputs. It also has a different name. IMO we should resolve this inconsistency before merging this patch: either both methods should close inputs, or neither should.

If both, we should make sure none of the call sites need the indexes to remain open.

If neither, I think it's good to keep the spirit of this patch (closing input indexes during hadoop indexing) by having the call site in IndexGeneratorJob do the closing.